lib/ansible/plugins/inventory/foreman.py

```diff
@@ -232,6 +247,14 @@ def _populate(self):
                 if self.get_option('want_facts'):
                     self.inventory.set_variable(host['name'], 'ansible_facts', self._get_facts(host))
 
+                # set Ansible groups from Foreman config groups
+                if self.get_option('use_config_groups'):
```

Comment:
instead of adding them directly why not expose as a variable and let user decide if they want to create groups based on it in 'compose'?

Reply:
Probably because it does an additional API call per host which makes it a lot slower.